### PR TITLE
Feature/issue 29/account login

### PR DESCRIPTION
**Pull request by yousfino**
Related Feature: #29 

- Implement login page UI
- Add tests for login page
- TODO: integration with backend

Login page UI:

image

### CODE REVIEW

**Comment by KhalilGarali** [comment]
Before merging this we might want to complete these first:
- use the logo file and not a screenshot for better quality
- adjust tailwind classes to better render the UI dynamically between different sizes of screens (Might need to meet with me Youssef so that register and login pages are coherent)

**Review by walidoow — COMMENTED** [review]
⚠️ Please use squash and merge here when you merge this PR @yousfino @NicolasChelico @KhalilGarali @JRB958 @MonikaaMoanes

**Review by KhalilGarali — CHANGES_REQUESTED** [review]

File: ClientApp/app/auth/login.tsx [hunk 1]
```diff
@@ -1,15 +1,143 @@
...
+      <View>
+        {/* Email Field */}
+        <View className="flex-row items-center bg-gray-100 rounded-3xl p-2 pl-4 min-h-16">
+          <MaterialCommunityIcons name="email" size={20} color="#aaa" />
+          <Controller
+            control={control}
+            name="email"
+            rules={{
+              required: "Email is required",
+              pattern: { value: /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/, message: "Invalid email" }
```
[the remark this answers is not in the document]
Comment (yousfino) [reply ClientApp/app/auth/login.tsx:44]: good catch! just fixed!

File: ClientApp/__tests__/auth/login.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    it('calls login function when valid data is entered', async () => {
+        const { getByPlaceholderText, getByText } = render(<LoginScreen />);
+
+        fireEvent.changeText(getByPlaceholderText('Email/username'), 'testuser@example.com');
+        fireEvent.changeText(getByPlaceholderText('Password'), 'testPassword');
+        fireEvent.press(getByText('Login'));
+
+        await waitFor(() => {
+            expect(Alert.alert).not.toHaveBeenCalled();
+        });
```
[the remark this answers is not in the document]
Comment (yousfino) [reply ClientApp/__tests__/auth/login.test.tsx:55]: agreed! done!

**Comment by yousfino** [comment]
This PR has been moved to another branch and has been resolved there due to a conflict with the current remote branch. Therefore, this PR had been closed without a merge.

Please click [here](https://github.com/SOEN490-SportsApp/SportsApp/pull/128) to view the revised version of this PR.

**Pull request closed without merging**